### PR TITLE
[BV-195] Run toolbox with the host user

### LINKED ISSUE #193: Enhancement | Run Toolbox as host user instead of root
## Describe the Enhancement

Based on [this issue](https://github.com/binbashar/leverage/issues/191), about files created inside the Leverage Toolbox container not being owned by the user running Leverage CLI (aka the host user), the following proposal was raised.

Instead of having an extra step inside the container doing a `chown` when finishing the command, the container itself can be run as the host user, thus creating the files owned by it.

E.g.
```shell
docker run -it --rm -u ${UID}:${GID} -v $(pwd):/app --workdir /app binbash/leverage-toolbox:1.2.7-0.1.9 version
```

Plus, this will contribute to security, avoiding the use of root inside the container.

## Expected Behavior

The user running inside the container should be the same as used on the host, then all the created files will be owned by it.

## Use Case

n/a

## Describe Ideal Solution

Run as mentioned before, the container using the host user.

## Alternatives Considered

Use a previously configured user (at build time in Dockerfile) and the run a chown.

## Additional Context

n/a

### PR DESCRIPTION
**Pull request by Franr**
## What?
We now run the commands in the toolbox with the same uid/gid of the host.
That remove many needs, like changing the ownership of files generated inside the container.

## Why?
* Simplification of logic in the CLI.
* Running containers as non-root user (securer).

## References
closes #193 

## Before release

Review the checklist [here](https://binbash.atlassian.net/l/cp/BthxSQ0j)
Integrations tests must be fixed (ref: https://github.com/binbashar/leverage/issues/272)

### CODE REVIEW

File: leverage/container.py [hunk 1]
```diff
@@ -381,61 +428,14 @@ def system_exec(self, command):
         self.entrypoint = self.AWS_CLI_BINARY
         return exit_code, output
 
-    def get_sso_code(self, container) -> str:
```
Comment (Franr) [inline leverage/container.py:384]: these were moved to the `SSOContainer` class, where they fit better

File: leverage/container.py [hunk 2]
```diff
@@ -381,61 +428,14 @@ def system_exec(self, command):
...
-
 
 class TerraformContainer(SSOContainer):
     """Leverage container specifically tailored to run Terraform commands.
     It handles authentication and some checks regarding where the command is being executed."""
 
     TF_BINARY = "/bin/terraform"
 
-    TF_MFA_ENTRYPOINT = "/root/scripts/aws-mfa/aws-mfa-entrypoint.sh"
-    TF_SSO_ENTRYPOINT = "/root/scripts/aws-sso/aws-sso-entrypoint.sh"
```
Comment (Franr) [inline leverage/container.py:438]: the SSO script was unused, is part of the CLI (in python) already

File: leverage/scripts/aws-mfa/aws-mfa-entrypoint.sh
Comment (Franr) [inline leverage/scripts/aws-mfa/aws-mfa-entrypoint.sh:1]: These were moved from the toolbox into the CLI, and are mounted into the container when we run commands.
That give us a better control and an easier way to introduce modifications over them.

File: leverage/modules/terraform.py [hunk 3]
```diff
@@ -323,23 +313,9 @@ def _init(tf, args):
 
     tf.paths.check_for_layer_location()
 
-    with LiveContainer(tf) as container:
```
Comment (Franr) [inline leverage/modules/terraform.py:326]: copying ssh keys and changing permissions is not required anymore, given now the user of the container and the host matches (thus permissions matches!)

File: leverage/containers/kubectl.py [hunk 4]
```diff
@@ -4,15 +4,15 @@
...
+from leverage._utils import AwsCredsEntryPoint, ExitError
 from leverage.container import TerraformContainer
 
 
 class KubeCtlContainer(TerraformContainer):
     """Container specifically tailored to run kubectl commands."""
 
     KUBECTL_CLI_BINARY = "/usr/local/bin/kubectl"
-    KUBECTL_CONFIG_PATH = Path("/root/.kube")
+    KUBECTL_CONFIG_PATH = Path("/home/leverage/.kube")
```
Comment (diego-ojeda-binbash) [inline leverage/containers/kubectl.py:15]: Please replace 'leverage' with the constant that defines that.

**Review by diego-ojeda-binbash — CHANGES_REQUESTED** [review]

File: leverage/containers/kubectl.py (hunk 4 above)
Comment (Franr) [reply leverage/containers/kubectl.py]: done :ok_hand:

File: Makefile [hunk 5]
```diff
@@ -1,7 +1,7 @@
 .PHONY: help build
 LEVERAGE_TESTING_IMAGE := binbash/leverage-cli-testing
 LEVERAGE_TESTING_TAG   := 2.5.0
-LEVERAGE_IMAGE_TAG     := 1.2.7-0.0.5
+LEVERAGE_IMAGE_TAG     := 1.2.7-0.1.14
```
[the remark this answers is not in the document]
Comment (Franr) [reply Makefile:4]: actually we now need the last toolbox image (0.1.19) that support the custom home folder

**Review by angelofenoglio — APPROVED** [review]

**Review by juanmatias — APPROVED** [review]

**Pull request merged**